### DISCUSSION TITLE
ReadableStream as a stub response?

### DISCUSSION THREAD

**Discussion by DennisKo** [cypress-io/cypress#28382] [Questions and Help]
Lets say I have a `ReadableStream` constructed with `const stream = new ReadableStream({ ... })`. Is it possible to do something like that?

```
const res = new Response(stream, {
      headers: {
        'Content-Type': 'text/event-stream',
      },
      statusCode: 200,
    });
cy.intercept('http://localhost:3001', res);
```

When I just try that I get `An invalid StaticResponse was supplied to cy.intercept(). headers must be a map of strings to strings.`

**Comment by portyu9** [comment]
Hi @DennisKo 

No — `cy.intercept()` does not currently accept a browser `Response` object or `ReadableStream` as a static response.

This:

```js
const res = new Response(stream, {
  headers: {
    'Content-Type': 'text/event-stream',
  },
})

cy.intercept('http://localhost:3001', res)
```

does not work because Cypress expects its own `StaticResponse` shape, not the Fetch API's `Response`.

Supported static bodies are:

- `object`
- `string`
- `ArrayBuffer`

For example:

```js
cy.intercept('GET', 'http://localhost:3001', {
  statusCode: 200,
  headers: {
    'content-type': 'text/event-stream',
  },
  body: 'data: first event\n\ndata: second event\n\n',
})
```

That can stub the **final body**, but it is not a real progressively delivered SSE stream.

If the behavior under test depends on timing/chunks from a real `ReadableStream` or `EventSource`, use a small test HTTP server that actually emits SSE/chunked data, then point the application at that server.

In other words:

- testing the result of an SSE response → a static `cy.intercept()` body may be enough;
- testing actual streaming/chunk timing/reconnection → use a real streaming server.

Please mark this answer as accepted if it helped, thank you.
